### PR TITLE
OSMO price and buy button in sidebar

### PR DESCRIPTION
**Pull request by JoseRFelix**

### CODE REVIEW

**Comment by syed8rahman** [comment]
Screen Shot 2022-12-05 at 4 24 13 PM
Not the exact PR but for side nav, the active section should be a different color.

**Comment by syed8rahman** [comment]
Screen Shot 2022-12-05 at 4 35 01 PM
Screen Shot 2022-12-05 at 4 35 16 PM

Buy button icons are in. Please animate a transition with 300ms Ease in out properties. 

Link to [Figma location here ](https://www.figma.com/file/a1FycJWJ7VARpWMfzoCQF1/UI-Refresh-Aug-2022?node-id=1%3A3&t=xGS9ronKniM3H2RZ-1)

**Comment by syed8rahman** [comment]
Screen Shot 2022-12-07 at 2 53 16 PM
There should be an inner padding of 8 px around the Osmo price + graph. 
We should also stick to 2 decimal places for consistency so it doesn't confuse users about rounding + accuracy in case it's .90 flat instead of .91 etc.

**Comment by syed8rahman** [comment]
Also looks like the value is top aligned with the osmo logo. Should be center aligned.

File: packages/web/components/skeleton.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import classNames from "classnames";
+import { DetailedHTMLProps, FunctionComponent } from "react";
+
+const Skeleton: FunctionComponent<
```
Comment (jonator) [inline packages/web/components/skeleton.tsx:4]: Could we name this something like `LoadingSkeletonPlaceholder` or something to clarify it's for loading data?

File: packages/web/components/navbar-osmo-price.tsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+      <Skeleton isLoaded={osmoPrice.isReady}>
+        <Button
+          mode="tertiary"
+          className={classNames(
+            "button group relative !h-11 gap-2 overflow-hidden !rounded-full !border-osmoverse-700 !py-1 font-bold text-osmoverse-100 !transition-all !duration-300 !ease-in-out",
+            "hover:border-none hover:bg-gradient-positive hover:text-osmoverse-1000"
+          )}
+          onClick={() => transferConfig.buyOsmo()}
+        >
+          <CreditCardIcon /> <span className="z-10">Buy tokens</span>{" "}
```
Comment (jonator) [inline packages/web/components/navbar-osmo-price.tsx:110]: Let's be sure to localize the "Buy Tokens" string

**Review by jonator — CHANGES_REQUESTED** [review]
Seems after closing the Transak modal, a weird padding is added to the bottom of the app. We may need to use some hack to prevent this by the Transak SDK. Ideally, the fix should be added to the Transak modal/integration code.

Screen Shot 2022-12-11 at 7 15 13 PM

File: packages/web/components/skeleton.tsx (hunk 1 above)
Comment (JoseRFelix) [reply packages/web/components/skeleton.tsx:4]: Renamed it into `SkeletonLoader`

**Comment by jonator** [comment]
It's worth noting that we also have a metric loader component that is similar. IMO we should eventually replace all metric loader usage with this new component.

https://github.com/osmosis-labs/osmosis-frontend/blob/e36f6b13b64bcd47353fb1830a56f2e450c0dca1/packages/web/components/loaders/metric-loader.tsx#L9

File: packages/web/integrations/transak/use-transak-modal.ts [hunk 3]
```diff
@@ -23,6 +23,8 @@ export function useTransakModal(
   const [transak, setTransak] = useState<any | null>(null);
   const [shouldShow, setShouldShow] = useState(false);
 
+  console.log(transak?.EVENTS);
```
Comment (jonator) [inline packages/web/integrations/transak/use-transak-modal.ts:26]: Please remove log

File: packages/web/components/navbar-osmo-price.tsx [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+      )}
+
+      {transferConfig?.fiatRampsModal && (
+        <FiatRampsModal {...transferConfig.fiatRampsModal} />
+      )}
+    </div>
+  );
+});
+
+const CreditCardIcon = () => {
```
Comment (jonator) [inline packages/web/components/navbar-osmo-price.tsx:138]: Let's move this into the components/assets folder

**Review by jonator — CHANGES_REQUESTED** [review]

File: packages/web/components/navbar-osmo-price.tsx (hunk 4 above)
Comment (JoseRFelix) [reply packages/web/components/navbar-osmo-price.tsx:138]: I was thinking about it, but the component has some deep CSS classes in the `path`s that only apply to this use case.

File: packages/web/integrations/transak/use-transak-modal.ts (hunk 3 above)
Comment (JoseRFelix) [reply packages/web/integrations/transak/use-transak-modal.ts:26]: Nice catch, thanks!

**Pull request merged**